src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -264,20 +264,28 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 		return exprStmt.WithExpression( result );
 	}
 
-	private InvocationExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+	private ExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+		ExpressionSyntax newExpr = invocationExpr;
 		var memberAccess = invocationExpr.Expression as MemberAccessExpressionSyntax;
+
 		if( string.Equals( memberAccess?.Name?.Identifier.ValueText, "ConfigureAwait", StringComparison.Ordinal ) ) {
 			if( memberAccess?.Expression is not null ) {
-				invocationExpr = (InvocationExpressionSyntax)memberAccess.Expression;
+				newExpr = memberAccess.Expression;
+				return Transform( newExpr );
 			}
 		}
 		return invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
 	}
 
-	private MemberAccessExpressionSyntax Transform( MemberAccessExpressionSyntax memberAccessExpr ) {
+	private ExpressionSyntax Transform( MemberAccessExpressionSyntax memberAccessExpr ) {
+		string[] identifiersToRemove = { "FromResult", "CompletedTask" };

```

Comment:
This does an allocation each call to `Transform`... another approach is to make a helper function that returns `bool` and does a `switch`, e.g.

```cs
bool MemberAccessesToRemove( MemeberAccessExpressionSyntax memberAccessExpr )
  => memberAccessExpr.Name.Identifier.ValueText switch {
    "FromResult" => true,
    "CompletedTask" => true,
    _ => false
  };
  ```
This general approach may also be more efficient than `.Contains` for a larger number of cases (currently the compiler turns a 7+-way switch into a code-gen'd hash table iirc). Not a signficant concern here of course.

Reply:
I wonder, however, if we should be more conservative here -- rather than making this call inside a member access expression, should we be doing it in a `ReturnStatementSyntax` (or whatever its called)? Like `return Task.FromResult( ... )` --> `return ...`.

I wonder because there is code like this:

```cs
var result = Task.CompletedTask;
return result;
```

which is not likely but I think we'd get maybe a bit further trying to syncify this than we'd like?